docs/contributing/adding_modules.md

```diff
@@ -319,6 +319,43 @@ function is defined as `processing_function` with all of the arguments before ru
 [`tqdm`][tqdm] package is leverage to give a progress bar and after completion the configuration file is written to file
 before a completion message is run.
 
+##### Results
+
+Before we start the parallel processing we create a dictionary `results = defaultdict()` which will have the results of
+processing added to. Some steps in processing return more than one object, for example `grainstats` returns statistics
+for the image along with height profiles. In such cases we need to instantiate a dictionary to hold each set of results
+across images and included a holder in the `for ... in pool.imap_unordered(...):` to store the results before adding the
```

Comment:
great to add documentation about this, thank you 👍

Reply:
Trying to put myself in the shoes of someone new coming to work on the project and not knowing how it all works[^1] and document these little things to help make it easier.

[^1]: This may include my future self!